### PR TITLE
Tengine lite update

### PR DESCRIPTION
**Pull request by liqi-c**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [ ] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: 3rdparty/libtengine/tengine.cmake [hunk 1]
```diff
@@ -62,24 +62,17 @@ ENDIF()
...
-	   endif()
-	else()
+	if(NOT ANDROID)
 		# linux system
 		if(CMAKE_SYSTEM_PROCESSOR STREQUAL arm)
-			   SET(CONFIG_ARCH_ARM32 ON)
+			   SET (TENGINE_TOOLCHIN_FLAG "-march=armv7-a")
 		elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL aarch64) ## AARCH64
-			   SET(CONFIG_ARCH_ARM64 ON)
+			   SET (TENGINE_TOOLCHIN_FLAG "-march=armv8-a")
```
Comment (alalek) [inline 3rdparty/libtengine/tengine.cmake:70]: > TOOLCHIN

typo

File: modules/dnn/src/dnn.cpp [hunk 2]
```diff
@@ -1611,7 +1620,32 @@ struct Net::Impl : public detail::NetImplBase
...
+
+        // Iterator to current layer.
+        MapIdToLayerData::iterator it = layers.begin();
+
+        for (; it != layers.end(); it++)
+        {
+            LayerData &ld = it->second;
+            Ptr<Layer> layerCur = ld.layerInstance;
 
+            if (!strcmp(ld.type.c_str(),"Convolution"))
```
Comment (alalek) [inline modules/dnn/src/dnn.cpp:1637]: Why `Layer::finalize()` doesn't work here?
(or just the first run of `Layer::forward()`, check integration of `convolutionOp` from OpenCL backend)

File: 3rdparty/libtengine/tengine.cmake (hunk 1 above)
Comment (liqi-c) [reply 3rdparty/libtengine/tengine.cmake]: done.

File: modules/dnn/src/dnn.cpp (hunk 2 above)
Comment (liqi-c) [reply modules/dnn/src/dnn.cpp]: Move to Layer::forward . please checkout again

File: 3rdparty/libtengine/tengine.cmake [hunk 3]
```diff
@@ -62,24 +62,17 @@ ENDIF()
...
-	   endif()
-	else()
+	if(NOT ANDROID)
 		# linux system
 		if(CMAKE_SYSTEM_PROCESSOR STREQUAL arm)
-			   SET(CONFIG_ARCH_ARM32 ON)
+			   SET (TENGINE_TOOLCHAIN_FLAG "-march=armv7-a")
 		elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL aarch64) ## AARCH64
-			   SET(CONFIG_ARCH_ARM64 ON)
+			   SET (TENGINE_TOOLCHAIN_FLAG "-march=armv8-a")
```
Comment (alalek) [inline 3rdparty/libtengine/tengine.cmake:70]: remove space before bracket.

File: modules/dnn/CMakeLists.txt [hunk 4]
```diff
@@ -131,11 +131,11 @@ endif()
 
 if(HAVE_TENGINE)
 	list(APPEND include_dirs ${TENGINE_INCLUDE_DIRS})
-	if(EXISTS ${TENGINE_LIBRARIES})
-		list(APPEND libs ${TENGINE_LIBRARIES})
+    if(EXISTS ${TENGINE_LIBRARIES})
+		list(APPEND libs -Wl,--whole-archive ${TENGINE_LIBRARIES} -Wl,--no-whole-archive)
```
Comment (alalek) [inline modules/dnn/CMakeLists.txt:135]: > if(EXISTS ${TENGINE_LIBRARIES})

Don't work with list as variable value.
Broken indentation.

No difference between both code paths. `ocv_add_dependencies()` is misused (required opencv modules) and does nothing.

**Review by wezeyf — APPROVED** [review]

File: 3rdparty/libtengine/tengine.cmake [hunk 5]
```diff
@@ -33,8 +32,9 @@ IF(EXISTS "${OCV_TENGINE_SOURCE_PATH}")
 	SET(BUILD_TENGINE ON)
 ELSE()
 	SET(OCV_TENGINE_FILENAME "${TENGINE_COMMIT_VERSION}.zip")#name2
-	SET(OCV_TENGINE_URL "https://github.com/OAID/Tengine/archive/") #url2
-	SET(tengine_md5sum f51ca8f3963faeeff3f019a6f6edc206) #md5sum2
+	#SET(OCV_TENGINE_URL "https://github.com/OAID/Tengine/archive/") #url2
+	SET(OCV_TENGINE_URL "https://github.com/liqi-c/Tengine/archive/") #url2.  Tmp url
```
Comment (alalek) [inline 3rdparty/libtengine/tengine.cmake:36]: You don't need that for testing. Upstream URL should work fine with commits from fork.

Comment (liqi-c) [reply 3rdparty/libtengine/tengine.cmake]: done.

File: modules/dnn/CMakeLists.txt (hunk 4 above)
Comment (liqi-c) [reply modules/dnn/CMakeLists.txt]: done.

**Review by alalek — APPROVED** [review]

**Pull request merged**